draft-ietf-quic-multipath.md

```diff
@@ -256,17 +279,52 @@ This extension does not change the definition of any transport parameter
 defined in {{Section 18.2. of QUIC-TRANSPORT}}.
 
 The transport parameter "active_connection_id_limit"
-{{QUIC-TRANSPORT}} limits the number of usable Connection IDs, and also
-limits the number of concurrent paths. However, endpoints might prefer to retain
-spare Connection IDs so that they can respond to unintentional migration events
-({{Section 9.5 of QUIC-TRANSPORT}}).
+{{QUIC-TRANSPORT}} limits the number of usable Connection IDs per path when the
+enable_multipath parameter is negotiated successfully. 
+Endpoints might prefer to retain spare Connection IDs so that they can 
+respond to unintentional migration events ({{Section 9.5 of QUIC-TRANSPORT}}). 
+
+The transport parameter "max_concurrent_paths" only becomes effective after the
+enable_multipath parameter is negotiated. Endpoints SHOULD use
+MP_NEW_CONNECTION_ID and MP_RETIRE_CONNECTION_ID frames to provide new Connection IDs 
+for the peer after the enable_multipath parameter is negotiated. 
+
+Path Identifier allocated by endpoints MUST NOT be larger or equal to 
+the max_concurrent_paths transport parameter which is negotiated by endpoints.
 
 Cipher suites with nonce shorter than 12 bytes cannot be used together with
 the multipath extension. If such cipher suite is selected and the use of the
 multipath extension is negotiated, endpoints MUST abort the handshake with a
 TRANSPORT_PARAMETER error.
 
 
+# Path Identifier {#pathid}
+
+The explicit Path Identifier is an integer between 0 and 2^62 - 1. 
+The Path Identifier is pre-allocated when endpoints provide new Connection IDs
+with MP_NEW_CONNECTION_ID frames {{mp-new-conn-id-frame}}.
+
+Each Connection ID is associated with a Path Identifier, as documented in {{mp-new-conn-id-frame}}. 
+Multiple connection IDs can be associated with the same path identifier.
+
+Endpoints use Path Identifier to address a path in the multi-path control frames,
+such as PATH_ABANDON, PATH_STANDBY, and PATH_AVAILABLE frames.
+
+Each endpoint associates a Receiver Packet Number space to each Path Identifier 
+that it provides to the peer. Each endpoint associates a Sender Packet Number space 
+to each Path Identifier received from the peer.
+
+The Path Identifier associated with the Destination Connection ID is used to 
+construct the packet protection nonce defined in {#multipath-aead}.
+
+The Path Identifier associated with the Destination Connection ID 
+is used to identify the path in ACK_MP frames {#ack-mp-frame}.
+
+Note that the Path Identifier for initial path is always 0. Connection IDs
+which are issued by origin NEW_CONNECTION_ID frames {{Section 19.15. of QUIC-TRANSPORT}}
+MUST be treated as their Path Identifier is 0.
+
```

Comment:
We also need to specify the path identifier for the connection ID specified in the "preferred address" transport parameter. Logically, it should be #1.

Reply:
Solved it.